### ISSUE TITLE
Error: foundSchema is null

### ISSUE THREAD

**Issue by maufl** [josdejong/jsoneditor#1158]
Hey, it looks like I was able to trigger a bug in `Node._findSchema`. I open a JSON file in the editor configured with a schema I provide. When I open child nodes in the JSON document, an error is thrown inside of jsoneditor. The error is `Error: foundSchema is null` in `Node.js:4451`.

This is a reduced version of the schema I'm using:
```json
{
    "$schema": "http://json-schema.org/draft-07/schema#",
    "type": "object",
    "patternProperties": {
        "^/[a-z0-9]*$": {
            "$ref": "#/definitions/component"
        }
    },
    "definitions": {
            "component": {
                "type": "object",
                "properties": {
                    "type": {
                        "type": "string",
                        "minLength": 1
                    },
                    "config": {
                        "type": "object"
                    },
                    "children": {
                        "type": "object",
                        "patternProperties": {
                            "^/[a-z0-9]+$": {
                                "$ref": "#/definitions/component"
                            }
                        }
                    }
                }
            }
    }
}
```

This is a reduced version of the JSON document I'm loading:
```json
{
    "/": {
        "type": "MessageOfTheDay"
    },
    "/status": {
        "type": "TabLayout",
        "children": {
            "/status":{
                "type": "StatusComponent",
                "config": {
                    "fields": []
                }
            },
            "/bus":{
                "type": "StatusComponent",
                "config": {
                    "fields": []
                }
            },
            "/system":{
                "type": "StatusComponent",
                "config": {
                    "fields": []
                }
            }
        }
    }
}
```

I tried to debug it but I'm so far unable to find the root error case. What I found out so far, is that JSON editor seems to not handle `$ref`s that point to a part of the same schema. This seems to be the reason why this error occurs when I expand the `children` node. I can work around it by explicitly  setting `schemaRefs: { "#/definitions/component": schema.definitions.component }` in the editor options. However I still get the error when expanding the `config` field.

One thing I noticed, is that `_findSchema` does iterate over the parts of the path to the node **and also** recursively call itself with sub paths to the node. This is also the reason why I don't really understand how `_findSchema` is supposed to work.

**Comment by josdejong** [comment]
Thanks for reporting and debugging this Felix. 

Sounds indeed like a bug in `Node._findSchema`. Anyone able to dig further to figure this out? Help would be welcome.

**Comment by maufl** [comment]
I found the original code hard to follow but I saw that there was a test suite for this method. I took the liberty of refactoring the whole method, I think it's a bit clearer now. I also added a test case for my problem and made sure that all tests are still green.

**Comment by josdejong** [comment]
> I found the original code hard to follow

yes definitely. I'm working on a completely new version of JSONEditor, the old codebase has grown too complicated.

**Comment by maufl** [comment]
This has been fixed with #1161

**Comment by josdejong** [comment]
Fix is now published in `v9.1.3`

**Comment by maufl** [comment]
I updated jsoneditor in our project and can confirm that the problem is solved for us :) Thanks again!
